src/basic/os-util.c

```diff
@@ -58,7 +58,47 @@ int path_is_extension_tree(const char *path, const char *extension) {
         return 1;
 }
 
-int open_extension_release(const char *root, const char *extension, char **ret_path, int *ret_fd) {
+static int check_extension_release_xattr(int extension_release_fd, char *extension_release_dir_path, char *filename) {
+        int r;
+
+        assert(extension_release_fd >= 0);
+        assert(extension_release_dir_path);
+        assert(filename);
+
+        /* No xattr or cannot parse it? Then skip this. */
+        _cleanup_free_ char *extension_release_xattr = NULL;
+        r = fgetxattr_malloc(extension_release_fd, "user.extension-release.strict", &extension_release_xattr);
+        if (r < 0) {
+                if (!ERRNO_IS_XATTR_ABSENT(r))
+                        log_debug_errno(r,
+                                        "%s/%s: Failed to read 'user.extension-release.strict' extended attribute from file, ignoring: %m",
+                                        extension_release_dir_path, filename);
+                else
+                        log_debug("%s/%s does not have user.extension-release.strict xattr, ignoring.", extension_release_dir_path, filename);
+
+                return r;
+        }
+
+        /* Explicitly set to request strict matching? Skip it. */
+        r = parse_boolean(extension_release_xattr);
```

Comment:
Maybe, it is better to introduce `fgetxattr_boolean()`.

Reply:
that seems overkill to me, it's used only here

Reply:
OK, let's do that in a later PR if it helps.